server/service/apple_mdm.go

```diff
@@ -2221,46 +2233,80 @@ func (svc *MDMAppleCheckinAndCommandService) TokenUpdate(r *mdm.Request, m *mdm.
 		}
 		if info.InstalledFromDEP {
```

Comment:
@gillespi314 @mna I'm thinking that everything inside this `if` should live in a cron job instead, that way if something goes wrong it's automatically retried.

What do you think? if you agree I can make an issue to try to tackle in this sprint.

Reply:
Sounds like a good idea. Are you thinking that we need a new cron or could we just add a new job to the worker queue?

Reply:
Makes sense to me. I think as a rule of thumb, we could say that if we need to call out third-party services that could be down independently of our infra (e.g. in this case the Apple DEP API), that's a good candidate for async cron/worker job with retries. Assuming there's no product expectation for immediate feedback for the happy path or something like that, of course (which I doubt there is in that case, given that the token update call is async itself, AIUI).

Reply:
Awesome, thanks! I created https://github.com/fleetdm/fleet/issues/11690